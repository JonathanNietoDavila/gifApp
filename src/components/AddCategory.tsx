import React, { useState } from 'react'

interface AddCategoryProps {
    setCategories?: Function
    onListenCategory: Function
}

export const AddCategory = ({ onListenCategory }: AddCategoryProps) => {
    const [inputValue, setInputValue] = useState<string>('')

    const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        setInputValue(e?.target?.value)
    }

    const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault()
        if (inputValue.trim().length <= 1) {
            return
        }
        onListenCategory(inputValue.trim())
        setInputValue('')
    }
    return (
        <form onSubmit={onSubmit}>
            {/* <input type='text' placeholder='Buscar gifs' value={inputValue} onChange={(e: ChangeEvent<HTMLInputElement>) => onInputChange(e)} /> */}
            <input type='text' placeholder='Buscar gifs' value={inputValue} onChange={onInputChange} />
        </form>
    )
}
