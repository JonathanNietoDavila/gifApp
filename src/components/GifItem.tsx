import React from 'react'

interface GifItemProps {
    id: string;
    title: string;
    url: string;
}
export const GifItem = ({ id, title, url }: GifItemProps) => {
    return (
        <section className='card'>
            <p>{title}</p>
            <img key={id} alt={title} src={url} />
        </section>
    )
}
