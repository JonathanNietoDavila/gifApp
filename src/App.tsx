import { useState } from 'react'
import { AddCategory, GifGrid } from './components/'


export const App = () => {
    const [categories, setCategories] = useState<string[]>(['Dragon Ball'])

    const onAddCategory = (category: string) => {
        if (categories.includes(category)) {
            return
        }
        setCategories(cts => [category, ...cts])
    }

    return (
        <>
            <h1>Gif App</h1>

            {/* <AddCategory setCategories={setCategories} /> */}
            <AddCategory onListenCategory={onAddCategory} />

            {
                categories.map((category: string) => (<GifGrid key={category} category={category} />))
            }
        </>
    )
}
