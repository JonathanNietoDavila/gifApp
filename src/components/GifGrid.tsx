import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifItem } from './GifItem'

interface GifGridProps {
    category: string
}

export const GifGrid = ({ category }: GifGridProps) => {
    const { gifs, isLoading } = useFetchGifs(category)

    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && (<h2>Cargando...</h2>)
            }
            <section className='card-grid'>
                {
                    gifs.map((gif) => (<GifItem key={gif.id} {...gif} />))
                }
            </section>
        </>
    )
}
