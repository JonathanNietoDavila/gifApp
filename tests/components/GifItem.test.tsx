import React from 'react';
import { render } from '@testing-library/react';

import { GifItem } from '../../src/components'

/**
 * @jest-environment jsdom
 */
describe('Pruebas en GifItem', () => {
    const id = 'pruebas-en-gif-item'
    const title = 'Pruebas en GifItem'
    const url = 'https://www.youtube.com/watch'
    test('should first', () => {
        const { container } = render(<GifItem id={id} title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })
})