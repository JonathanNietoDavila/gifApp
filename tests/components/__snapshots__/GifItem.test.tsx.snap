// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem should first 1`] = `
<div>
  <section
    class="card"
  >
    <p>
      Pruebas en GifItem
    </p>
    <img
      alt="Pruebas en GifItem"
      src="https://www.youtube.com/watch"
    />
  </section>
</div>
`;
